src/main/java/com/beecoders/ras/service/OrderService.java

```diff
@@ -77,7 +79,13 @@ public Long save(AddOrderRequest addOrderRequest, String table){
             orderDishes.add(orderDish);
         }
         orderDishRepository.saveAll(orderDishes);
-        order.setTotalPrice(order.getTotalPrice()+orderDishes.stream().map(OrderDish::getPrice).reduce(Double::sum).orElse(0D));
+        order.setTotalPrice(order.getTotalPrice()+orderDishes.stream()
+                .map(OrderDish::getPrice)
+                .reduce(Double::sum)
+                .orElse(0D));
+
+        if (!Objects.isNull(order.getPromocode()))
+            calculateOrderPrice(order);
 
```

Comment:
I think in `else` case we need to set `current_sum` field with `total_price` value because of this information will be returned as API response for order